src/Debugger.cpp

```diff
@@ -945,6 +948,99 @@ void Debugger::setupUi() {
 
 	ui.setupUi(this);
 
+#if 1
+	auto splitter = new QSplitter(this);
+	splitter->setOrientation(Qt::Vertical);
+
+	{
+		logger_ = new QPlainTextEdit(this);
+		logger_->setReadOnly(true);
+		QFont font("monospace");
+		font.setStyleHint(QFont::TypeWriter);
+		logger_->setFont(font);
+		logger_->setWordWrapMode(QTextOption::WrapMode::NoWrap);
+		logger_->setHorizontalScrollBarPolicy(Qt::ScrollBarAsNeeded);
+		logger_instance = logger_;
+
+		qInstallMessageHandler([](QtMsgType type, const QMessageLogContext &, const QString &message) {
+			if (!logger_instance) {
+				return;
```

Comment:
When does this happen?

Reply:
Not very often, it's protection against if the code is not before the logger is setup.

Admittedly, that doesn't happen because we simply don't have any debug messages that early... But better to be safe I suppose 😉

Reply:
My point was that according to this code, this never happens. The handler is only installed via `qInstallMessageHandler()` right after `logger_instance` is assigned.

Reply:
Yeah, I actually realized that while thinking about it on my drive into the office this morning. You are 100% right, it literally CAN'T happen. May as well delete the dead code.